packages/contracts/test/manager/facets/ManagerFacet.t.sol

```diff
@@ -2,101 +2,191 @@
+        assertEq(IManagerFacet.getMasterChefAddress(), contract1);
     }
 
     function testShouldSetFormulasAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setFormulasAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getFormulasAddress(), contract1);
+        IManagerFacet.setFormulasAddress(contract1);
+        assertEq(IManagerFacet.getFormulasAddress(), contract1);
     }
 
     function testShouldSetBondingShareAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setBondingShareAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getBondingShareAddress(), contract1);
+        IManagerFacet.setBondingShareAddress(contract1);
+        assertEq(IManagerFacet.getBondingShareAddress(), contract1);
     }
 
     function testShouldSetStableSwapMetaPoolAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setStableSwapMetaPoolAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getStableSwapMetaPoolAddress(), contract1);
+        IManagerFacet.setStableSwapMetaPoolAddress(contract1);
+        assertEq(IManagerFacet.getStableSwapMetaPoolAddress(), contract1);
     }
 
     function testShouldSetBondingContractAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setBondingContractAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getBondingContractAddress(), contract1);
+        IManagerFacet.setBondingContractAddress(contract1);
+        assertEq(IManagerFacet.getBondingContractAddress(), contract1);
     }
 
     function testShouldSetTreasuryAddress() public prankAs(admin) {
-        ManagerFacet(address(diamond)).setTreasuryAddress(contract1);
-        assertEq(ManagerFacet(address(diamond)).getTreasuryAddress(), contract1);
+        IManagerFacet.setTreasuryAddress(contract1);
+        assertEq(IManagerFacet.getTreasuryAddress(), contract1);
     }
 
     function testShouldSetIncentiveToUAD() public prankAs(admin) {
         address dollarTokenAddress = generateAddress("dollarTokenAddress", true, 10 ether);
-        ManagerFacet(address(diamond)).setDollarTokenAddress(dollarTokenAddress);
-        ManagerFacet(address(diamond)).setIncentiveToUAD(user1, contract1);
+        IManagerFacet.setDollarTokenAddress(dollarTokenAddress);
+        IManagerFacet.setIncentiveToUAD(user1, contract1);
+    }
+
+    function testShouldSetMinterRoleWhenInitializing() public prankAs(admin) {
+        assertEq(IManagerFacet.hasRole(UBQ_MINTER_ROLE, admin), true);
+    }
+
+    function testShouldInitializeDollarTokenAddress() public prankAs(admin) {
+        assertEq(IManagerFacet.getDollarTokenAddress(), address(diamond));
+    }
+
+    function testShouldDeployStableSwapPool() public {
+
+        vm.startPrank(admin);
+
+        MockuADToken uAD;
+        MockUbiquityGovernance uGov;
+
+        uAD = new MockuADToken(10000);
+        uGov = new MockUbiquityGovernance(10000);
```

Comment:
Pretty funny that you're using `uGov` as this is a deprecated way to refer to our governance token. Perhaps you can keep all of these variables as references like `dollarToken` `governanceToken` `creditToken` ?